server/statistics/hot_cache.go

```diff
@@ -17,11 +17,17 @@ import (
 	"context"
 
 	"github.com/tikv/pd/server/core"
+	"go.uber.org/atomic"
 )
 
 // Denoising is an option to calculate flow base on the real heartbeats. Should
 // only turned off by the simulator and the test.
-var Denoising = true
+var Denoising = atomic.NewBool(true)
```

Comment:
I think it's unnecessary to use an atomic variable?

Reply:
there may be race if no atomic.

Reply:
we met data race before